lib/Driver/UnixToolChains.cpp

```diff
@@ -171,7 +171,10 @@ toolchains::GenericUnix::constructInvocation(const DynamicLinkJobAction &job,
   }
 
   if (Linker.empty()) {
-    Linker = getDefaultLinker();
+    if (getTriple().isAndroid())
+      Linker = "lld";
+    else
+      Linker = getDefaultLinker();
```

Comment:
Since those platforms cannot use the NDK tools (they can use the sysroot as you point out, but not the binaries), switching the linker for them is not going to work, because they cannot run it. Using the system linker for them is the best bet. In any case, I think the decision of which linker to use should be contained in a function, not scattered in the middle of a bigger function.

Reply:
Got it, will update.